src/main/java/de/tum/in/www1/artemis/web/rest/AccountResource.java

```diff
@@ -91,7 +89,7 @@ private boolean isSAML2Disabled() {
      * {@code POST  /register} : register the user.
      *
      * @param managedUserVM the managed user View Model.
-     * @throws InvalidPasswordException {@code 400 (Bad Request)} if the password is incorrect.
+     * @throws InvalidPasswordException  {@code 400 (Bad Request)} if the password is incorrect.
```

Comment:
I think this should be an 401 instead - https://datatracker.ietf.org/doc/html/rfc7235#section-3.1

Reply:
Not my exception, the naming is a bit ambiguous: This exception gets thrown if the provided password doesn't match the requirements (registration, etc.)

Changing it would by out of scope in my opinion.

Reply:
Ok - I agree that this is out of scope for this feature. Maybe we should open an improvement issue for that